### PR TITLE
Implements #384: Youtube Feeds without API

### PR DESCRIPTION
**Pull request by XIncognito10**
Implemented api-less youtube feeds (#384)
users, playlists and channels are supported

### CODE REVIEW

**Comment by XIncognito10** [comment]
There you go, let me know if you need anything else

**Comment by elelay** [comment]
Code is good.
Testing it more I noticed that the channel image is not retrieved when subscribing to e.g. https://www.youtube.com/user/sixguntv/playlists in the new code.
This is because there is no channel id for `get_real_cover()` in `youtube.py`.
It could maybe be fixed by modifying `get_real_cover()` but I noticed that the feed from e.g. https://www.youtube.com/feeds/videos.xml?user=sixguntv does contain the channel id.
Using your method to resolve the feed, then fetching and extracting `yt:channelId` when subscribing would be better, what do you think ?

Excerpt from the feed:
```xml
<feed xmlns:yt="http://www.youtube.com/xml/schemas/2015" xmlns:media="http://search.yahoo.com/mrss/" xmlns="http://www.w3.org/2005/Atom">
 <link rel="self" href="http://www.youtube.com/feeds/videos.xml?user=sixguntv"/>
 <id>yt:channel:UCaITiY0rLVsRr71dP-kQgEg</id>
 <yt:channelId>UCaITiY0rLVsRr71dP-kQgEg</yt:channelId>
 <title>SixgunTV</title>
```

**Comment by XIncognito10** [comment]
Yeah, I'd agree. It seems it would be more reliable since all the feeds, would come with that id

**Comment by elelay** [comment]
Yes, with pleasure, if you're willing to ;-)
You can use `util.urlopen` to load the feed, and maybe `xml.etree.ElementTree.parse` to parse the feed?

**Comment by XIncognito10** [comment]
There you go, i modified get_real_cover() to use the new api. I also took the liberty of replacing the regular expression with a dedicated html parser, but you can let me know if you would like to revert the changes to the latter

File: src/gpodder/youtube.py [hunk 1]
```diff
@@ -209,26 +203,44 @@ def return_user_feed(url, channel):
...
+            channel_url = 'https://www.youtube.com/channel/{}'.format(channel_id)
+            html_data = util.urlopen(channel_url).read().decode('utf-8')
+            parser = YouTubeHTMLCoverParser()
+            parser.feed(html_data)
+            if parser.url:
+                logger.debug('Youtube cover art for {} is: {}'.format(url, parser.url))
+                return parser.url
 
-    return for_each_feed_pattern(return_user_cover, url, None)
+        except Exception as e:
```
Comment (elelay) [inline src/gpodder/youtube.py]: no need for `as e`

**Comment by XIncognito10** [comment]
im not sure i follow

**Comment by XIncognito10** [comment]
Do you mean like converting all the feeds to their respective channels? If so i didnt do it because with the current setup, the user can have playlist subscriptions different from user subscriptions different from channel subscriptions, instead of all of them being channel subscriptions. I think thats more flexible. Is that what you meant?

**Comment by elelay** [comment]
My bad, I thought each playlist, user, channel would get a different /channelId/. Let me investigate some more...

**Comment by XIncognito10** [comment]
sure! no problem

**Comment by elelay** [comment]
Merged, thanks!

**Pull request merged**